Firestore/Source/Local/FSTLocalStore.h

```diff
@@ -181,7 +182,7 @@ NS_ASSUME_NONNULL_BEGIN
 - (model::DocumentMap)executeQuery:(FSTQuery *)query;
 
 /** Notify the local store of the changed views to locally pin / unpin documents. */
-- (void)notifyLocalViewChanges:(NSArray<FSTLocalViewChanges *> *)viewChanges;
+- (void)notifyLocalViewChanges:(std::vector<local::LocalViewChanges> &&)viewChanges;
```

Comment:
This method should take `viewChanges` by const reference, not by rvalue reference.

Rvalue references in the signature mean that ownership is transferred. If you have a method that takes ownership of the given parameter, and you want to enforce that the parameter _must_ be moved, then it makes sense to declare the argument as an rvalue reference. If the method only needs to observe the given parameter (only call `const` methods on it), it should take it by const reference.

The simple guideline is: only use rvalue references when ownership transfer takes place, otherwise pass by const reference or by pointer, just like in C++03.

You _could_ make `notifyLocalViewChanges` more efficient by taking ownership of `viewChanges` and moving `added`/`removed_keys` out of it, but that would very much go into overkill territory. The existing code wasn't doing that.

Reply:
I see. Thanks. Fixed.